bindinfo/handle.go

```diff
@@ -596,34 +596,39 @@ func (h *BindHandle) logicalDeleteBindInfoSQL(originalSQL, db string, updateTs t
 // CaptureBaselines is used to automatically capture plan baselines.
 func (h *BindHandle) CaptureBaselines() {
 	parser4Capture := parser.New()
-	schemas, sqls := stmtsummary.StmtSummaryByDigestMap.GetMoreThanOnceBindableStmt()
-	for i := range sqls {
-		stmt, err := parser4Capture.ParseOneStmt(sqls[i], "", "")
+	bindableStmts := stmtsummary.StmtSummaryByDigestMap.GetMoreThanOnceBindableStmt()
+	for _, bindableStmt := range bindableStmts {
+		stmt, err := parser4Capture.ParseOneStmt(bindableStmt.Query, "", "")
 		if err != nil {
-			logutil.BgLogger().Debug("parse SQL failed", zap.String("SQL", sqls[i]), zap.Error(err))
+			logutil.BgLogger().Debug("parse SQL failed", zap.String("SQL", bindableStmt.Query), zap.Error(err))
 			continue
 		}
 		if insertStmt, ok := stmt.(*ast.InsertStmt); ok && insertStmt.Select == nil {
 			continue
 		}
-		normalizedSQL, digest := parser.NormalizeDigest(sqls[i])
-		dbName := utilparser.GetDefaultDB(stmt, schemas[i])
+		normalizedSQL, digest := parser.NormalizeDigest(bindableStmt.Query)
+		dbName := utilparser.GetDefaultDB(stmt, bindableStmt.Schema)
 		if r := h.GetBindRecord(digest, normalizedSQL, dbName); r != nil && r.HasUsingBinding() {
 			continue
 		}
-		h.sctx.Lock()
-		h.sctx.GetSessionVars().CurrentDB = schemas[i]
-		oriIsolationRead := h.sctx.GetSessionVars().IsolationReadEngines
-		// TODO: support all engines plan hint in capture baselines.
-		h.sctx.GetSessionVars().IsolationReadEngines = map[kv.StoreType]struct{}{kv.TiKV: {}}
-		hints, err := getHintsForSQL(h.sctx.Context, sqls[i])
-		h.sctx.GetSessionVars().IsolationReadEngines = oriIsolationRead
-		h.sctx.Unlock()
-		if err != nil {
-			logutil.BgLogger().Debug("generate hints failed", zap.String("SQL", sqls[i]), zap.Error(err))
-			continue
+		var hints string
+		if bindableStmt.Prepared {
+			hints = bindableStmt.PlanHint
+		} else {
+			h.sctx.Lock()
+			h.sctx.GetSessionVars().CurrentDB = bindableStmt.Schema
+			oriIsolationRead := h.sctx.GetSessionVars().IsolationReadEngines
+			// TODO: support all engines plan hint in capture baselines.
+			h.sctx.GetSessionVars().IsolationReadEngines = map[kv.StoreType]struct{}{kv.TiKV: {}}
+			hints, err = getHintsForSQL(h.sctx.Context, bindableStmt.Query)
+			h.sctx.GetSessionVars().IsolationReadEngines = oriIsolationRead
+			h.sctx.Unlock()
+			if err != nil {
+				logutil.BgLogger().Debug("generate hints failed", zap.String("SQL", bindableStmt.Query), zap.Error(err))
+				continue
+			}
```

Comment:
Can we generate hints for non-prepared stmts in `getEncodedPlan`  as we do for prepared stmts?